### PR TITLE
feat: chat feature

### PR DESCRIPTION
**Pull request by Rendisaputra33**
## Summary by CodeRabbit

- **New Features**
	- Introduced enhanced chat functionalities, including conversation management, message handling, and AI-driven summarization capabilities.
	- Added permissions for audio recording and Bluetooth functionality to support new features.
	- Implemented a speech-to-text feature for improved user interaction.
	- Integrated Algolia search capabilities for lecturer accounts.
	- Added a user interface for summarizing chat conversations.
	- Created a data model for managing chat summaries within Firestore.
	- Enhanced configuration management with environment variables.

- **Bug Fixes**
	- Addressed various state management issues to ensure smoother navigation and user experience.

- **Documentation**
	- Updated `pubspec.yaml` to include new dependencies that enhance chat, search, and speech recognition capabilities.
	- Added `.env.example` file to guide developers in setting up environment variables for Algolia.

- **Chores**
	- Removed unnecessary configuration entries and improved project structure for better maintainability.

### CODE REVIEW

File: lib/services/algolia.dart [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import 'dart:async';
+
+import 'package:algoliasearch/algoliasearch_lite.dart';
+import 'package:signsyncai/features/auth/domain/account.dart';
+
+Future<List<Account>> searchLecture(String? query) async {
+  final client = SearchClient(
+    appId: 'WJ636HPOTV',
+    apiKey: '457068778f317d016fe598dfc69d99df',
```
[the remark this answers is not in the document]
Comment (yarabramasta) [reply lib/services/algolia.dart:9]: apiKey check https://pub.dev/packages/flutter_dotenv. lk mari tak update e ndek pipeline e.

File: lib/features/chats/presentation/widget/speach_to_text.dart
Comment (yarabramasta) [inline lib/features/chats/presentation/widget/speach_to_text.dart:1]: typo speech_to_text

File: lib/services/firebase.dart
Comment (yarabramasta) [inline lib/services/firebase.dart:1]: gemini prompt e ngga efektif tah lk bhs inggris expect bhs indo?

Comment (yarabramasta) [reply lib/services/firebase.dart:1]: sebenere ono seh json i18n e, cuman gpp lk te digae full bhs indo, in the end sg nggae yo mahasiswa kro dosen wkwkw

File: tugas.md
Comment (yarabramasta) [inline tugas.md:1]: tugas.md?

**Review by yarabramasta — COMMENTED** [review]
leave open in response to coderabbit review

File: tugas.md
Comment (Rendisaputra33) [reply tugas.md:1]: sorry cuy wingi seleh tugas kuliah dek project lali ra tak hapus

File: lib/services/firebase.dart
Comment (Rendisaputra33) [reply lib/services/firebase.dart:1]: sejauh iki tak cobak cobak efektifan indo se, btw lek chate gawe english resulte bakal english sisan

Comment (yarabramasta) [reply lib/services/firebase.dart:1]: wokee no prob indo ae

**Comment by yarabramasta** [comment]
Soal CI/CD. Codemagic ternyata ws ono step gae setup env. So, implementasi load dotenv e ngga via file. Aku mari check sisan, pkg flutter_dotenv default e load system env + .env file lk options e empty. Diupdate ae implementasi ne kyok ngene:

https://github.com/yarabramasta/clinic/blob/f8c6261fc72e2397a2826688c8f64eef72fd402d/lib/main.dart#L16

https://github.com/yarabramasta/clinic/blob/f8c6261fc72e2397a2826688c8f64eef72fd402d/lib/services/http.dart#L10

**Pull request merged**